engine/src/test/java/com/datatorrent/stram/StramLocalClusterTest.java

```diff
@@ -274,4 +286,103 @@ public WindowGenerator setupWindowGenerator()
     localCluster.shutdown();
   }
 
+  @Test
+  public void testDynamicLoading() throws IOException, ClassNotFoundException
+  {
+    final String generatedJar = generatejar();
+    File file = new File(generatedJar);
+    final Class<?> pojo = URLClassLoader.newInstance(new URL[]{file.toURI().toURL()}).loadClass("POJO");
+
+    StreamingApplication app = new StreamingApplication()
+    {
+      @Override
+      public void populateDAG(DAG dag, Configuration conf)
+      {
+        TestGeneratorInputOperator genNode = dag.addOperator("genNode", TestGeneratorInputOperator.class);
+        genNode.setMaxTuples(2);
+
+        DynamicLoader dynamicLoader = dag.addOperator("DynamicLoader", new DynamicLoader());
+        dynamicLoader.setClass("POJO");
+
+        dag.addStream("fromNode1", genNode.outport, dynamicLoader.in);
+
+        dag.setInputPortAttribute(dynamicLoader.in, Context.PortContext.TUPLE_CLASS, pojo);
+        conf.set(LIBJARS_CONF_KEY_NAME, generatedJar);
+      }
+    };
+
+    LocalMode.runApp(app, 10000);
```

Comment:
Please structure the unit tests so that you wait for expected result instead of waiting for arbitrary time. There are examples in Malhar for this.